pkg/dns/config/sync.go

```diff
@@ -130,18 +145,34 @@ func (sync *kubeSync) processUpdate(result syncResult, buildUnchangedConfig bool
 	return
 }
 
-func (sync *kubeSync) updateFederations(data map[string]string, config *Config) (err error) {
-	if flagValue, ok := data["federations"]; ok {
-		config.Federations = make(map[string]string)
-		if err = fed.ParseFederationsFlag(flagValue, config.Federations); err != nil {
-			glog.Errorf("Invalid federations value: %v (value was %q)",
-				err, data["federations"])
-			return
-		}
-		glog.V(2).Infof("Updated federations to %v", config.Federations)
-	} else {
-		glog.V(2).Infof("No federations present")
+type fieldUpdateFn func(key string, data string, config *Config) error
+
+func updateFederations(key string, value string, config *Config) error {
+	config.Federations = make(map[string]string)
+	if err := fed.ParseFederationsFlag(value, config.Federations); err != nil {
+		glog.Errorf("Invalid federations value: %v (value was %q)", err, value)
+		return err
 	}
+	glog.V(2).Infof("Updated federations to %v", config.Federations)
 
-	return
+	return nil
+}
+
+func updateStubDomains(key string, value string, config *Config) error {
+	config.StubDomains = make(map[string][]string)
+	if err := json.Unmarshal([]byte(value), &config.StubDomains); err != nil {
+		glog.Errorf("Invalid JSON: %q", value)
+		return err
+	}
+
```

Comment:
Should log updateStubDomains() and updateUpstreamNameservers() just like updateFederations().